### PR TITLE
Add output_preamble to match postamble

### PR DESCRIPTION
**Pull request by CrookedGrin**
### What are you trying to accomplish?

Add output_preamble to match existing output_postamble, as suggested in [this issue](https://github.com/ViewComponent/view_component/issues/1931).

### What approach did you choose and why?

To keep this PR as simple as possible, we've just mirrored the existing output_postamble code.

### Anything you want to highlight for special attention from reviewers?

This is a follow-up to [another PR](https://github.com/ViewComponent/view_component/pull/1932), incorporating the changes suggested there.

### CODE REVIEW

File: lib/view_component/base.rb [hunk 1]
```diff
@@ -104,11 +104,16 @@ def render_in(view_context, &block)
...
+        rendered_template = safe_render_template_for(@__vc_variant).to_s
+
+        if output_preamble.blank? && output_postamble.blank?
+          rendered_template
         else
-          safe_render_template_for(@__vc_variant).to_s + safe_output_postamble
+          preamble = output_preamble.blank? ? "" : safe_output_preamble
+          postamble = output_postamble.blank? ? "" : safe_output_postamble
+
+          preamble + rendered_template + postamble
```
Comment (camertron) [inline lib/view_component/base.rb:116]: `#maybe_escape_html` already checks for blankness, so I think this code can be simplified a bit:

```suggestion
          safe_output_preamble + rendered_template + safe_output_postamble
```

**Review by camertron — CHANGES_REQUESTED** [review]
Nice, thanks!

**Comment by camertron** [comment]
Not sure what's going on with this PVC failure, will investigate.

File: lib/view_component/base.rb (hunk 1 above)
Comment (CrookedGrin) [reply lib/view_component/base.rb]: Hey @camertron, I accepted your suggestion, but it looks like the build then started failing on the Primer tests. Any idea why this change could have done that?

Comment (camertron) [reply lib/view_component/base.rb]: Yeah, not sure, will look into it this afternoon 👍

**Comment by camertron** [comment]
Ok, the failure is happening on PVC main too, so this PR shouldn't be blocked.

**Pull request merged**